app/controllers/api/v1/stats/channels_controller.rb

```diff
@@ -0,0 +1,65 @@
+class Api::V1::Stats::ChannelsController < Api::BaseController
+
+  def channels
+
+    if(params[:uuid] == nil)
+
+      channels = Channel.all.map { |channel| channel.id }
+      data = JSON.pretty_generate(channels)
+      render(json: data)
+
+    else
+
+      channel = Channel.find_by_id(params[:uuid])
+
+      if(channel == nil)
+
+        error_response = JSON.pretty_generate({
+          errors: [{
+            status: "404",
+            title: "Not Found",
+            detail: "Channel with uuid " + params[:uuid] + " not found"
+          }]
+        })
+        render(json: error_response)
+
+      else
+
+        case channel.details_type
+
+          when "SiteChannelDetails"
+            channel_details = SiteChannelDetails.find_by_id(channel.details_id)
+            channel_type = "website"
+            channel_name = channel_details.url
+          when "YoutubeChannelDetails"
+            channel_details = YoutubeChannelDetails.find_by_id(channel.details_id)
+            channel_type = "youtube"
+            channel_name = channel_details.name
+          when "TwitterChannelDetails"
+            channel_details = TwitterChannelDetails.find_by_id(channel.details_id)
+            channel_type = "twitter"
+            channel_name = channel_details.name
+          when "TwitchChannelDetails"
+            channel_details = TwitchChannelDetails.find_by_id(channel.details_id)
+            channel_type = "twitch"
+            channel_name = channel_details.name
+          end
+
+          data = JSON.pretty_generate({
+            uuid: channel.id,
+            channel_id: channel_details.channel_identifier,
+            channel_type: channel_type,
```

Comment:
Can we use this helper that has already been created instead of defining our own `channel_type` variable in each one of these?
https://github.com/brave-intl/publishers/blob/666ea3ea39a7aaeb9face944413d3685bf62a605/app/helpers/publishers_helper.rb#L367

Reply:
Normally I would agree, but since this is an api for stats I wouldn't want `I18n.t` to rename values in unpredictable ways. Certainly will want to use this if we ever render this data to the browser